### PR TITLE
feat: HUBSA-2362 create action for configure aws profile

### PR DESCRIPTION
**Pull request by rbrisuda**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+name: continuous-integration
+on:
+  pull_request:
+    branches: [main]
+jobs:
+  conventional-commits:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
```
Comment (mmichna) [inline .github/workflows/ci.yml:9]: @lucky-edu: this is very expensive GH job/step definition - much better is merge all step to one GH workflow (pricing is round to minutes per every GH job)

File: action.yml [hunk 2]
```diff
@@ -0,0 +1,82 @@
+name: Configure AWS Profile
+
+description: Configures an OIDC AWS Role Session as an aws config profile.
+
+
+inputs:
+  role-to-assume:
+    description: "ARN of the role to assume."
```
Comment (mmichna) [inline action.yml:8]: PLS remove "." from all descriptions. thx

File: action.yml [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    required: false
+  aws-session-token:
+    description: "AWS Session Token"
+    required: false
+  aws-source-profile:
+    description: "Name of source profile for assuming role by SDKs"
+    required: false
+
+outputs:
+  aws-profile:
```
Comment (mmichna) [inline action.yml:35]: maybe is better use same like input `aws-profile-name`

File: action.yml [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    description: "Created AWS profile name"
+    value: ${{ inputs.aws-profile-name || github.action }}
+
+runs:
+  using: "composite"
+  steps:
+    - name: Setup Profile Based on Source Profile
+      shell: bash
+      if: inputs.aws-source-profile
```
Comment (mmichna) [inline action.yml:44]: `inputs.aws-source-profile == false`???

File: action.yml [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+        PROFILE_NAME=${{ inputs.aws-profile-name || github.action }}
+        
+        # Set profile for assuming role by SDK
+
+        aws configure set role_arn ${{ inputs.role-to-assume }} --profile ${PROFILE_NAME}
+        aws configure set source_profile ${{ inputs.aws-source-profile }} --profile ${PROFILE_NAME}   
+        aws configure set region ${{ inputs.aws-region }} --profile ${PROFILE_NAME}    
+
+    - name: Configure AWS Credentials
+      uses: aws-actions/configure-aws-credentials@v2
```
Comment (mmichna) [inline action.yml:55]: all inputs are automatically mask by this action - maybe is better disable masking `mask-aws-account-id: false`

File: action.yml [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+    value: ${{ inputs.aws-profile-name || github.action }}
+
+runs:
+  using: "composite"
+  steps:
+    - name: Setup Profile Based on Source Profile
+      shell: bash
+      if: inputs.aws-source-profile
+      run: |
+        PROFILE_NAME=${{ inputs.aws-profile-name || github.action }}
```
Comment (mmichna) [inline action.yml:46]: maybe move **PROFILE_NAME** to action env and then use same value instead of local variable - also in output

**Review by mmichna — APPROVED** [review]

File: action.yml (hunk 4 above)
Comment (rbrisuda) [reply action.yml:44]: this is correct, so if:  inputs.aws-source-profile is not empty, then it is evaluated as TRUE so it sets only profile based on source profile

File: action.yml (hunk 3 above)
Comment (rbrisuda) [reply action.yml:35]: like discussed in previous pr, this naming is on purpose, so name for name which is not creatred yet, without name, created profile

File: action.yml (hunk 6 above)
Comment (rbrisuda) [reply action.yml:46]: can be improved later, I need to have some version merged where to start with

File: action.yml (hunk 2 above)
Comment (rbrisuda) [reply action.yml:8]: I will do that in next pr

File: .github/workflows/ci.yml (hunk 1 above)
Comment (rbrisuda) [reply .github/workflows/ci.yml:9]: @mmichna pls don't write these type of commnents into PRs where all resolved conversations are required to pass checks

**Review by jcajkovic — APPROVED** [review]

**Pull request merged**